### PR TITLE
Allow static files with non-ASCII characters

### PR DESCRIPTION
**Pull request by tusbar**
See https://github.com/IONISx/edx-theme/issues/40

### CODE REVIEW

**Comment by tusbar** [comment]
And https://github.com/django/django/blob/731f313d604a6cc141f36d8a1ba9a75790c70154/django/contrib/staticfiles/storage.py#L237

**Comment by Kelketek** [comment]
This seems to have fixed static asset collection for me. Looking at the code, this appears to be backwards compatible-- while the code assumes the file will be unicode, the previous code would have assumed ASCII. As UTF-8 is forward compatible with ASCII, this should not break anyone's code it previously worked for, but may fix things for certain edge cases, like the one I was running into.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @tusbar! I've created [OSPR-592](https://openedx.atlassian.net/browse/OSPR-592) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. 

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ('this must be merged by XX date', and why that is)
- partner information ('this is a course on edx.org')
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

We can't start reviewing your pull request until you've added yourself to the [AUTHORS](https://github.com/edx/django-staticfiles/blob/master/AUTHORS) file. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

**Comment by sarina** [comment]
@Kelketek cool. I'm going to ask the Platform team to take a look as well.

File: staticfiles/storage.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def post_process(self, paths, dry_run=False, fail_silently=False):
 
                 # ..to apply each replacement pattern to the content
                 if name in adjustable_paths:
-                    content = original_file.read()
+                    content = original_file.read().decode('utf8')
```
Comment (cpennington) [inline staticfiles/storage.py]: How do you know the file contents are `utf-8`?

Comment (cpennington) [inline staticfiles/storage.py]: (And if they aren't, does this raise a readable error in a reasonable place?)

Comment (cpennington) [inline staticfiles/storage.py]: Lastly, does upstream django-staticfiles handle this?

Comment (tusbar) [inline staticfiles/storage.py]: https://github.com/django/django/blob/731f313d604a6cc141f36d8a1ba9a75790c70154/django/contrib/staticfiles/storage.py#L237

They use `settings.FILE_CHARSET`, which defaults to 'utf-8'. I guess we could use that as it's in fact available in django 1.4. 

I'll update the PR.

**Comment by tusbar** [comment]
@cpennington done.

**Comment by tusbar** [comment]
If the file contents cannot be decoded with `settings.FILE_CHARSET`, it’ll throw an error like [this one](https://github.com/IONISx/edx-theme/issues/400) upon collection.

**Comment by cpennington** [comment]
:+1:

**Comment by cpennington** [comment]
@tusbar I'm going to merge this, will you submit a PR to edx-platform to update the pinned version there?

**Comment by cpennington** [comment]
I'm going to go ahead and just bump the version. Thanks for the patch, @tusbar!

**Pull request merged**